File: frontend/lib/shared/models/complaint_model.dart
class ComplaintModel {
  final int? id;
  final String title;
  final String description;
  final String status;
  final String? location;
  final String? photo;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final DateTime? resolvedAt;
  final int? userId;

  ComplaintModel({
    this.id,
    required this.title,
    required this.description,
    this.status = 'pending',
    this.location,
    this.photo,
    this.createdAt,
    this.updatedAt,
    this.resolvedAt,
    this.userId,
  });

  factory ComplaintModel.fromJson(Map<String, dynamic> json) {
    return ComplaintModel(
      id: json['id'] as int?,
      title: json['title'] as String,
      description: json['description'] as String,
      status: json['status'] as String? ?? 'pending',
      location: json['location'] as String?,
      photo: json['photo'] as String?,
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
      updatedAt: json['updated_at'] != null ? DateTime.parse(json['updated_at']) : null,
      resolvedAt: json['resolved_at'] != null ? DateTime.parse(json['resolved_at']) : null,
      userId: json['user_id'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'status': status,
      'location': location,
      'photo': photo,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
      'resolved_at': resolvedAt?.toIso8601String(),
      'user_id': userId,
    };
  }
}

File: backend/src/modules/complaint/complaint.model.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/db');
const User = require('../auth/auth.model');

const Complaint = sequelize.define('Complaint', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  status: {
    type: DataTypes.STRING,
    defaultValue: 'pending'
  },
  location: {
    type: DataTypes.STRING
  },
  photo: {
    type: DataTypes.STRING
  },
  in_progress_at: {
    type: DataTypes.DATE,
    allowNull: true,
    defaultValue: null
  },
  resolved_at: {
    type: DataTypes.DATE,
    allowNull: true,
    defaultValue: null
  },
  user_id: {
    type: DataTypes.INTEGER,
    references: {
      model: User,
      key: 'id'
    }
  }
}, {
  tableName: 'complaints',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at'
});

// Associations
User.hasMany(Complaint, { foreignKey: 'user_id' });
Complaint.belongsTo(User, { foreignKey: 'user_id' });

module.exports = Complaint;
